src/lib/rest/rest.cpp

```diff
@@ -193,6 +195,23 @@ static int uriArgumentGet(void* cbDataP, MHD_ValueKind kind, const char* ckey, c
       ciP->uriParamTypes.push_back(val);
     }
   }
+  else if (key == URI_PARAM_Q)
+  {
+    if (*val != 0)
```

Comment:
> I can't really see any reason to move the parse part to Scope ...

The main reason is uniformity/homogeneity. From an external looker or a newcomer to the code it is difficult to explain why some filters are implemented as part of Scope (in particular the ones related with geo-stuff) and others (the one related with q) are implemented in a different way.